ffs/manager/manager.go

```diff
@@ -85,8 +88,40 @@ func (m *Manager) Create(ctx context.Context) (ffs.APIID, string, error) {
 	defer m.lock.Unlock()
 
 	log.Info("creating instance")
+
+	var addr address.Address
+	if m.ffsUseMasterAddr {
+		addr = m.wm.MasterAddr()
+	} else {
+		res, err := m.wm.NewAddress(ctx, "bls")
+		if err != nil {
+			return ffs.EmptyInstanceID, "", fmt.Errorf("creating new wallet addr: %s", err)
+		}
+		a, err := address.NewFromString(res)
+		if err != nil {
+			return ffs.EmptyInstanceID, "", fmt.Errorf("decoding newly created addr: %s", err)
+		}
+		addr = a
+	}
+
+	addrInfo := api.AddrInfo{Name: "Initial Address", Addr: addr.String()}
+	switch addr.Protocol() {
```

Comment:
Kind of regretting having `AddInfo.Type`, but to not break that API, I'm detecting the address type here, so in the case of a master address passed in from Lotus, we still have the `Type` field set. 

In retrospect, and maybe for v2 APIs, I think we should keep addresses as `address.Address` everywhere possible. It's a nice type, like `cid.Cid`, that serializes and has some useful methods on it.

Reply:
Yeah. Actually if v2 also includes supporting other clients than Lotus, then the _types_ problem is much broader since `address.Address` is a Lotus specific type. 
We can think of re-using Lotus specific type as the base type for Powergate... but that's hard to know.

Reply:
`address.Address` is in `github.com/filecoin-project/go-address`, so maybe not lotus specific?